listenbrainz/db/timescale.py

```diff
@@ -21,6 +21,9 @@ def init_db_connection(connect_str):
     for more info.
     """
     global engine
+    if not connect_str:
```

Comment:
Is this guard also needed? In [`init_timescale_connection`](https://github.com/metabrainz/listenbrainz-server/blob/8677e14353ab9efecd361d5ad4dff4b3db4cfc01/listenbrainz/webserver/timescale_connection.py#L7) we also have this guard.

It's odd to see that we appear to call init_db_connection twice - once in [`gen_app`](https://github.com/metabrainz/listenbrainz-server/blob/bbf6a832b30a7e3fa1b35f808299646d75719c32/listenbrainz/webserver/__init__.py#L112) and again in the [constructor for `TimescaleListenStore`](https://github.com/metabrainz/listenbrainz-server/blob/e3c0d66b9dede36bee02a7429d645f52b2b422c7/listenbrainz/listenstore/timescale_listenstore.py#L52). I don't think that both of these calls have to be here - if we didn't call it in gen_app then this guard wouldn't be needed.

Not important to fix for this release, I'll open a ticket

Reply:
I believe its better to keep this guard as well, for non-web timescale connections that might be attempted. As for the double init, lets do a ticket.